File: framework/languages/java/e2e/SpotService/Shared/src/main/java/systems/zlink/e2e/spotservice/shared/NoopIngressHandler.java
package systems.zlink.e2e.spotservice.shared;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

import systems.zlink.framework.ZLinkMessageContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;

public final class NoopIngressHandler
    implements ZLinkRequestHandler<Contracts.StateReq, Contracts.StateRes> {
    private final ScenarioState state;

    public NoopIngressHandler(ScenarioState state) {
        this.state = state;
    }

    @Override
    public CompletionStage<Contracts.StateRes> handle(
        Contracts.StateReq request,
        ZLinkMessageContext context) {
        state.record("IngressReq", "channel", request.op());
        return CompletableFuture.completedFuture(
            new Contracts.StateRes("", state.nodeRid(), request.op()));
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Server/Play/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/play/handlers/NoopIngressHandler.kt
package systems.zlink.e2e.kotlin.spotservice.play.handlers

import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.ScenarioState
import systems.zlink.e2e.kotlin.spotservice.play.spots.*
import systems.zlink.framework.ZLinkMessageContext
import systems.zlink.framework.kotlin.ZLinkSuspendingRequestHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup

@ZLinkHandlerGroup(Contracts.CHANNEL_HANDLER_GROUP)
class NoopIngressHandler(
    private val state: ScenarioState,
) : ZLinkSuspendingRequestHandler<Contracts.StateReq, Contracts.StateRes> {
    override suspend fun handle(
        request: Contracts.StateReq,
        context: ZLinkMessageContext,
    ): Contracts.StateRes {
        state.record("IngressRequest", "channel", request.op)
        return Contracts.StateRes("", state.nodeRid(), request.op)
    }
}
